great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py

```diff
@@ -3156,104 +3390,136 @@ def _filter(domain: Domain) -> bool:
+            ):
+                expectation_plot_impl = self._get_expect_domain_values_ordinal_chart
+            elif DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=quantitative_metrics
+            ):
+                expectation_plot_impl = (
+                    self._get_expect_domain_values_to_be_between_chart
+                )
+            elif DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=temporal_metrics
+            ):
+                expectation_plot_impl = self._get_expect_domain_values_temporal_chart
 
-            return plot_impl(
+            return expectation_plot_impl(
                 expectation_type=expectation_type,
                 df=df,
-                metric_name=metric_name,
+                sanitized_metric_names=sanitized_metric_names,
                 sequential=sequential,
                 subtitle=subtitle,
             )
         else:
-            plot_impl: Optional[
+            metric_plot_impl: Optional[
                 Callable[
                     [
                         pd.DataFrame,
-                        str,
+                        Set[str],
                         bool,
                         Optional[str],
                     ],
-                    alt.Chart,
+                    Union[alt.Chart, alt.LayerChart],
                 ]
             ] = None
 
-            if metric_name in nominal_metrics:
-                plot_impl = self._get_nominal_metric_chart
-            elif metric_name in ordinal_metrics:
-                plot_impl = self._get_ordinal_metric_chart
-            elif metric_name in quantitative_metrics:
-                plot_impl = self._get_quantitative_metric_chart
-            elif metric_name in temporal_metrics:
-                plot_impl = self._get_temporal_metric_chart
+            if DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=nominal_metrics
+            ):
+                metric_plot_impl = self._get_nominal_metrics_chart
+            elif DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=ordinal_metrics
+            ):
+                metric_plot_impl = self._get_ordinal_metrics_chart
+            elif DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=quantitative_metrics
+            ):
+                metric_plot_impl = self._get_quantitative_metrics_chart
+            elif DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=temporal_metrics
+            ):
+                metric_plot_impl = self._get_temporal_metrics_chart
 
-            return plot_impl(
+            return metric_plot_impl(
```

Comment:
Similarly `metric_plot_impl` could be None.

Reply:
✅